13.0/s2i/bin/assemble

```diff
@@ -171,7 +178,43 @@ function configure_proxy() {
   xml="$xml\
        </proxy>"
     sed -i "s|<!-- ### configured http proxy ### -->|$xml|" $HOME/.m2/settings.xml
+    return
   fi
+
```

Comment:
Couldn't you replace the whole if block above this with a couple small additions to your new code and just have a single path?